File: benchmarks/NSubstitute.Analyzers.Benchmarks.Source.CSharp/DiagnosticsSources/NonSubstitutableMemberArgumentMatcherDiagnosticsSource.cs
using NSubstitute.Analyzers.Benchmarks.Source.CSharp.Models;

namespace NSubstitute.Analyzers.Benchmarks.Source.CSharp.DiagnosticsSources
{
    public class NonSubstitutableMemberArgumentMatcherDiagnosticsSource
    {
        public void NS1004_ArgumentMatcherUsedWithNonVirtualMember()
        {
            var substitute = Substitute.For<Foo>();
            substitute.ObjectReturningMethodWithArguments(Arg.Any<int>(), Arg.Compat.Any<int>(), Arg.Is(1m));
            substitute.ObjectReturningMethodWithArguments(Arg.Do<int>(i => { }), Arg.Compat.Do<int>(i => { }), Arg.Compat.Is(1m));
            substitute.ObjectReturningMethodWithArguments(Arg.Invoke(1), Arg.InvokeDelegate<int>(), Arg.Compat.InvokeDelegate<int>());
            substitute.ObjectReturningMethodWithArguments(Arg.Compat.Invoke(1), Arg.Compat.Do<int>(i => { }), Arg.Compat.Is(1m));
            substitute.InternalObjectReturningMethodWithArguments(Arg.Is(1));
            substitute.InternalObjectReturningMethodWithArguments(Arg.Compat.Is(1));
            substitute.InternalObjectReturningMethodWithArguments(Arg.Any<int>());
            substitute.InternalObjectReturningMethodWithArguments(Arg.Compat.Any<int>());
            _ = substitute[Arg.Any<int>()];
            _ = substitute[Arg.Compat.Any<int>()];
        }
    }
}

File: benchmarks/NSubstitute.Analyzers.Benchmarks.Source.VisualBasic/DiagnosticsSources/NonSubstitutableMemberArgumentMatcherDiagnosticsSource.vb
Imports System
Imports NSubstitute.Analyzers.Benchmarks.Source.VisualBasic.Models

Namespace DiagnosticsSources
    Public Class NonSubstitutableMemberArgumentMatcherDiagnosticsSource
        Public Sub NS1004_ArgumentMatcherUsedWithNonVirtualMember()
            Dim substitute = NSubstitute.Substitute.[For] (Of Foo)()
            substitute.ObjectReturningMethodWithArguments(Arg.Any (Of Integer)(), Arg.Compat.Any (Of Integer)(),
                                                          Arg.[Is](1D))
            substitute.ObjectReturningMethodWithArguments(
                Arg.[Do] (Of Integer)(Function(i)
                    Throw New Exception
                End Function),
                Arg.Compat.[Do] (Of Integer)(Function(i)
                    Throw New Exception
                End Function), Arg.Compat.[Is](1D))
            substitute.ObjectReturningMethodWithArguments(Arg.Invoke(1), Arg.InvokeDelegate (Of Integer)(),
                                                          Arg.Compat.InvokeDelegate (Of Integer)())
            substitute.ObjectReturningMethodWithArguments(Arg.Compat.Invoke(1),
                                                          Arg.Compat.[Do] (Of Integer)(Function(i)
                                                              Throw New Exception
                                                          End Function), Arg.Compat.[Is](1D))
            substitute.InternalObjectReturningMethodWithArguments(Arg.[Is](1))
            substitute.InternalObjectReturningMethodWithArguments(Arg.Compat.[Is](1))
            substitute.InternalObjectReturningMethodWithArguments(Arg.Any (Of Integer)())
            substitute.InternalObjectReturningMethodWithArguments(Arg.Compat.Any (Of Integer)())
            Dim first = substitute(Arg.Any (Of Integer)())
            Dim second = substitute(Arg.Compat.Any (Of Integer)())
        End Sub
    End Class
End Namespace
